x-pack/elastic-agent/pkg/capabilities/rule.go

```diff
@@ -0,0 +1,112 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package capabilities
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	allowKey     = "allow"
+	denyKey      = "deny"
+	conditionKey = "__condition__"
+)
+
+type ruler interface {
+	Rule() string
```

Comment:
What is 'Rule' supposed to return? Maybe we should name the method `RuleType()`?

Why do we need the return type "string" here? AFAICT the only valid values are `allow` and `deny`. If valid values are restricted we should introduce an enum.

I don't see the method `Rule()` to be used anywhere? Do you plan to use it somewhere in the future?